### ISSUE TITLE
Release build of USD on Win10 will not execute usdview, but debug build of USD works fine.

### ISSUE THREAD

**Issue by elbrandt** [PixarAnimationStudios/USD#960]
### Description of Issue
Release build of USD on Win10 will not execute usdview, but debug build of USD works fine.

### Steps to Reproduce
I built usd on my Win10 machine using build_usd.py from the current state of the git repo.  The build completed successfully, and I believe I set all necessary paths properly.

When I run usdview from the command line, I type 'usdview <my-valid-usda-file.usda>' and press return. The command prompt pauses for about a second, then returns me to the command prompt with no error message issued (and no usd view window showing). 
If I launch a python2.7 interpreter and type 'import pxr.Usdviewq', I am kicked out of the python interpreter, back to the command prompt, with no errors issued. I tried attaching the VS debugger to a running python27.exe and repeating this, but the VS debugger does not catch any exceptions - the process just terminates when I issue 'import pxr.Usdviewq'

The strange thing is, if I build the build_usd.py with the --debug flag, then usdview works, and so does python command 'import pxr.Usdviewq' 

How can I troubleshoot this so I can understand why my release build of usd fails, but my debug build succeeds? I am very familiar with VS build environment & tools, but not particularly familiar with QT nor pyside  nor automatically generate python bindings in general, so I'm not sure how to troubleshoot.

The only non-standard thing I did when building (both non-working Release and working Debug) was to use VS2019 with the -Tv141 flag passed to CMake. This causes VS to use the VS2017 toolset, rather than the v142 (VS2019) toolset, and should be identical to building with VS2017, as the usd build instructions suggest.

I am aware of one other colleague who has tried the same steps as me, and is having this identical same problem with a release build, so while this case may be peculiar, it is not unique.

Thanks!
-Eric.

### System Information (OS, Hardware)
Win10.0.17134 Build 17134
Core i7-8850H
NVidia Quadro P2000

### Package Versions
VS2019 (building with v141 (aka VS2017) toolset
C:\WINDOWS\system32>pip list
Package    Version
---------- -------
pip        19.2.3
PyOpenGL   3.1.0
PySide     1.2.4
setuptools 40.6.2
wheel      0.33.6

### Build Flags
CMake flag: -Tv141
No flags supplied to build_usd.py -> problem described above exists
--debug flag supplied to build_usd.py -> problem described above doesn't exist.

**Comment by sunyab** [comment]
Hi @elbrandt, I haven't used VS2019 before and I'm having a hard time setting things up so that the v141 toolset is used. Did you modify build_usd.py to pass in -Tv141 when invoking CMake? 

If you could attach the log.txt file that should be in your <inst>/build/USD directory that might help me try to reproduce the issue. Thanks!

**Comment by elbrandt** [comment]
Hi @sunyab,

Yes, I made small modifications to build_usd.py to pass -Tv141 so I could build with VS2019. using the VS2017 toolset. 

I have attached 3 files:
1. my modified build_usd.py (with changed extension to allow attachment to this msg)
2. the build log from my release build (that exhibits the problem) and
3. the build log from my --debug build (that executes usdview just fine).
I hope this helps you diagnose the problem. Let me know what else I can do to help.
-Eric.

[build_usd_elb.py.txt](https://github.com/PixarAnimationStudios/USD/files/3612887/build_usd_elb.py.txt)
[log_debug_build.txt](https://github.com/PixarAnimationStudios/USD/files/3612889/log_debug_build.txt)
[log_release_build.txt](https://github.com/PixarAnimationStudios/USD/files/3612890/log_release_build.txt)

**Comment by sunyab** [comment]
Thanks for the info! I had to modify your version of the script further to get a build going (including updating the version of boost to 1.71, since 1.65 didn't build with VS2019) but I believe I am able to reproduce the issue over here. 

- If I just do 'from pxr import Gf' or 'from pxr import Tf' I see the silent crash behavior. 
- If I try to run sdfdump.exe on any USD file I get an error that looks like:

```
(TfDiagnosticType)1: in _CreateResolver at line 158 of C:\Projects\USD\pxr\usd\lib\ar\resolver.cpp -- Invalid resolver type

(TfDiagnosticType)1: in _RegisterFormatPlugins at line 189 of C:\Projects\USD\pxr\usd\lib\sdf\fileFormatRegistry.cpp -- Failed verification: ' !formatBaseType.IsUnknown() '

(TfDiagnosticType)1: in _OpenLayerAndUnlockRegistry at line 3010 of C:\Projects\USD\pxr\usd\lib\sdf\layer.cpp -- Cannot determine file format for @shot.usd@

sdfdump: Error - failed to open layer <shot.usd>
```

Can you confirm whether you're seeing the same thing?

**Comment by elbrandt** [comment]
Hi @sunyab,

Yes, I can confirm the exact same behavior in sdfump on my release build. See below.

(Strangely, I don't recall having to make any changes to my boost. I received some #pragma warnings when building that came from boost saying my compiler was untested (or similar) but it gave no errors building with 1.65, in fact my boost output dlls in my output lib directory are all of the form boost-*-vc141-mt-1_65_1.dll)

Thanks for continuing to investigate this!
-Eric

SDF Test:
`
D:\buildtools\usd\bin>sdfdump test.usda
(TfDiagnosticType)1: in _CreateResolver at line 158 of D:\buildtools\USD_src\pxr\usd\lib\ar\resolver.cpp -- Invalid resolver type
(TfDiagnosticType)1: in _RegisterFormatPlugins at line 189 of D:\buildtools\USD_src\pxr\usd\lib\sdf\fileFormatRegistry.cpp -- Failed verification: ' !formatBaseType.IsUnknown() '
(TfDiagnosticType)1: in _OpenLayerAndUnlockRegistry at line 3010 of D:\buildtools\USD_src\pxr\usd\lib\sdf\layer.cpp -- Cannot determine file format for @test.usda@
sdfdump: Error - failed to open layer <test.usda>
`

**Comment by ChokmahBinah** [comment]
Hi there,

I had the same problem.
I don't know if it helps, but after setting Python27 to path all worked like a charm here =$

**Comment by mattstoneham** [comment]
![Capture](https://user-images.githubusercontent.com/5821941/77706565-a7522480-6fba-11ea-92ce-8963a6ef18eb.PNG)

I had the *exact* same issue (albeit building in VS17 rather than 19). Turns out I was missing the VS2017 component highlighted here.

**Comment by sunyab** [comment]
Closing this old issue out. USD 20.05 has a number of changes to better support building with Visual Studio 2019; this includes a fix for #1095 which seems to be the likely cause for the original issue. Thanks!
